tests/typecheck/utils/test_functional.yml

```diff
@@ -1,23 +1,31 @@
 -   case: cached_property_class_vs_instance_attributes
     main: |
       from django.utils.functional import cached_property
-      from typing import List
+      from typing import List, ClassVar
 
       class Foo:
           @cached_property
           def attr(self) -> List[str]: ...
-          @cached_property  # E: Argument 1 to "cached_property" has incompatible type "Callable[[Foo, str], List[str]]"; expected "Callable[[Any], List[str]]"  [arg-type]
+
+          @cached_property  # E: Too many arguments for property  [misc]
           def attr2(self, arg2: str) -> List[str]: ...
 
-          reveal_type(attr)      # N: Revealed type is "django.utils.functional.cached_property[builtins.list[builtins.str]]"
-          reveal_type(attr.name) # N: Revealed type is "Union[builtins.str, None]"
+          reveal_type(attr)  # N: Revealed type is "def (self: main.Foo) -> builtins.list[builtins.str]"
 
-      reveal_type(Foo.attr)      # N: Revealed type is "django.utils.functional.cached_property[builtins.list[builtins.str]]"
-      reveal_type(Foo.attr.func) # N: Revealed type is "def (Any) -> builtins.list[builtins.str]"
+      reveal_type(Foo.attr)  # N: Revealed type is "def (self: main.Foo) -> builtins.list[builtins.str]"
 
       f = Foo()
-      reveal_type(f.attr)        # N: Revealed type is "builtins.list[builtins.str]"
-      f.attr.name                # E: "List[str]" has no attribute "name"  [attr-defined]
+      reveal_type(f.attr)  # N: Revealed type is "builtins.list[builtins.str]"
+      f.attr.func  # E: "List[str]" has no attribute "func"  [attr-defined]
+
+      # May be overridden by @property
+      class Bar(Foo):
+          @property
+          def attr(self) -> List[str]: ...
+
+      # May be overridden by ClassVar
+      class Quux(Foo):
+          attr: ClassVar[List[str]] = []
```

Comment:
I don't know if it's worth testing/mentioning but it doesn't work the other way around.
If the base class use ClassVar, sublcasses cannot override with `cached_property` / `property` (even tho django does that sometimes.

See https://mypy-play.net/?mypy=1.6.0&python=3.11&flags=strict&gist=fe1b4709168b54b0cb3dc83b07874292